File: LeetcodeQuestions/beautiful_arrangements_ii-667.py
class Solution(object):
    def constructArray(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[int]
        """
        res = [None] * n
        low = 1
        high = n
        idx = 0
        is_high = False

        res[idx] = low
        idx += 1
        low += 1

        while k > 1:
            res[idx] = high
            idx += 1
            high -= 1
            k -= 1
            is_high = True

            if k > 1:
                res[idx] = low
                idx += 1
                low += 1
                k -= 1
                is_high = False

        while idx < n:
            if is_high:
                res[idx] = high
                high -= 1
            else:
                res[idx] = low
                low += 1
            idx += 1
        return res

File: LeetcodeQuestions/beautiful_arrangements_ii-667.rb
# @param {Integer} n
# @param {Integer} k
# @return {Integer[]}
def construct_array(n, k)
    res = Array(n)
    idx = 0
    min = 1
    max = n
    is_max = false
    res[idx] = min
    idx += 1
    min += 1
    
    while k > 1
        res[idx] = max
        idx += 1
        max -= 1
        is_max = true
        k -= 1
        
        if k > 1
            res[idx] = min
            idx += 1
            min += 1
            k -= 1
            is_max = false
        end
    end
    
    while idx < n
        if is_max
            res[idx] = max
            idx += 1
            max -= 1
        else
            res[idx] = min
            idx += 1
            min += 1
        end
    end
    res
end